### ISSUE TITLE
Version 4.2.4 unable to display characters

### ISSUE THREAD

**Issue by ody999** [utmapp/UTM#5238]
**Describe the issue**
UTM 4.2.4(beta) won't display characters on ChromiumBrowser, VScode when GPU acceleration is enabled (ANGLE metal backened).
Besides, still chances are that Ubuntu 20.04 will freeze randomly(screen display freezes, no response to keyboard, sometimes can move the mouse, but sometimes not) when selecting display cards with "GPU supported". Do not know this issue belongs to Ubuntu itself of UTM GPU accleration.

**Configuration (required)**
* UTM Version: V4.2.4
* OS Version: Ubuntu 20.04.5
* Device Model: ipadpro 2021 M1 256GB
* Is it jailbroken (name jailbreak used)?  No
* How did you install UTM?  TrollStore

**syslog before freeze**
Apr 24 11:28:55 ody-QEMU-Virtual-Machine systemd[1]: Started Hostname Service.
Apr 24 11:29:01 ody-QEMU-Virtual-Machine systemd[1367]: tracker-extract.service: Succeeded.
Apr 24 11:29:16 ody-QEMU-Virtual-Machine systemd[1367]: Started Application launched by gnome-shell.
Apr 24 11:29:18 ody-QEMU-Virtual-Machine gnome-shell[1591]: Some code accessed the property 'discreteGpuAvailable' on the module 'appDisplay'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Apr 24 11:29:19 ody-QEMU-Virtual-Machine gnome-shell[1591]: ../clutter/clutter/clutter-actor.c:10558: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
Apr 24 11:29:21 ody-QEMU-Virtual-Machine tracker-store[2454]: OK
Apr 24 11:29:21 ody-QEMU-Virtual-Machine systemd[1367]: tracker-store.service: Succeeded.
Apr 24 11:29:25 ody-QEMU-Virtual-Machine systemd[1]: systemd-hostnamed.service: Succeeded.
Apr 24 11:29:47 ody-QEMU-Virtual-Machine systemd-timesyncd[552]: Timed out waiting for reply from 240.0.0.44:123 (ntp.ubuntu.com).
Apr 24 11:29:58 ody-QEMU-Virtual-Machine kernel: [  169.801100] kauditd_printk_skb: 5 callbacks suppressed
Apr 24 11:29:58 ody-QEMU-Virtual-Machine kernel: [  169.801114] audit: type=1400 audit(1682306998.168:81): apparmor="DENIED" operation="open" profile="snap.chromium.chromium" name="/proc/2046/mem" pid=2004 comm="chrome_crashpad" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/112958.171944:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/112958.186351:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/112958.187624:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/112958.187760:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/112958.188127:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/112958.225625:ERROR:elf_dynamic_array_reader.h(64)] tag not found
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/112958.325498:WARNING:exception_snapshot_linux.cc(263)] fpsimd not found
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/112958.326697:ERROR:file_io_posix.cc(144)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory (2)
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/112958.326713:ERROR:file_io_posix.cc(144)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: No such file or directory (2)
Apr 24 11:29:58 ody-QEMU-Virtual-Machine gnome-shell[1907]: [1907:1907:0424/112958.352389:ERROR:gpu_process_host.cc(942)] GPU process exited unexpectedly: exit_code=512
Apr 24 11:30:01 ody-QEMU-Virtual-Machine CRON[2621]: (root) CMD ([ -x /etc/init.d/anacron ] && if [ ! -d /run/systemd/system ]; then /usr/sbin/invoke-rc.d anacron start >/dev/null; fi)
Apr 24 11:30:13 ody-QEMU-Virtual-Machine kernel: [  185.031879] audit: type=1400 audit(1682307013.396:82): apparmor="DENIED" operation="open" profile="snap.chromium.chromium" name="/proc/2618/mem" pid=2004 comm="chrome_crashpad" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/113013.402507:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/113013.402962:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/113013.404128:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/113013.404299:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/113013.404334:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/113013.438667:ERROR:elf_dynamic_array_reader.h(64)] tag not found
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/113013.447345:WARNING:exception_snapshot_linux.cc(263)] fpsimd not found
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/113013.447534:ERROR:file_io_posix.cc(144)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory (2)
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[2004]: [0424/113013.447559:ERROR:file_io_posix.cc(144)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: No such file or directory (2)
Apr 24 11:30:13 ody-QEMU-Virtual-Machine gnome-shell[1907]: [1907:1907:0424/113013.454478:ERROR:gpu_process_host.cc(942)] GPU process exited unexpectedly: exit_code=512

![Screenshot from 2023-04-24 10-56-33](https://user-images.githubusercontent.com/118160587/233892924-49efb8cb-73cd-4c48-9e85-355bb5a4f87f.png)
![Screenshot from 2023-04-24 11-03-18](https://user-images.githubusercontent.com/118160587/233893001-45d891e3-d050-4928-9c05-ff61f7347685.png)

**Comment by osy** [comment]
Please check your settings to see what rendering backend is selected. If it is not "Default", can you try setting it to Default?

**Comment by ody999** [comment]
Tried "Default" and "ANGLE OpenGL", both let UTM crashed soon after log in to the system.

**Comment by osy** [comment]
Did you try ANGLE Metal as well?

**Comment by ody999** [comment]
Yes, I used angle(metal) first, UTM won’t crash but there’s a display problem for chromium, and the system could freeze after some time( as described above). Then I tried Default and OpenGL, but UTM crashed quickly even before I can open an application.

**Comment by osy** [comment]
Does the OpenGL/Default issues happen for you in 4.1.x? I know the Metal issues are new to 4.2.4, but was OpenGL broken before as well?

**Comment by ody999** [comment]
Sorry, I only used metal for v4.1.x, and everything works fine except for system freezing problem. Oh，plus，the screen may flash sometime（sudden black screen and recovered within 0.5 second），maybe it's due to too high resolution I set for VM（same reso as ipadpro itself）.

**Comment by osy** [comment]
Can you test v4.1.6 with OpenGL just to see if the issues are new or existing?

**Comment by osy** [comment]
By the way, I believe the crash is due to OpenGL backend using more memory. You can try decreasing the amount of memory to the VM and see if that helps (other way is to lower the resolution).

**Comment by osy** [comment]
Can you test this build? https://drive.google.com/file/d/17hsVMmIsX77coB_PRwzJ5vOpLAPeWu9f/view?usp=share_link

**Comment by ody999** [comment]
For v4.1.6, both Default and OpenGL woluld crash like v4.2.4. Indeed, when using ANGLE OpenGL, before crash, I'll get the UTM message for "running in low memory" while Default will crash directly(I set 2GB and 4GB ram, both crash).
Thanks for this new build, characters now can display normally!!
Plus, I found that the system is less likely to freeze when I terminate other iOS apps manually and only run UTM, but I still wonder whether it's related to GPU acceleration and can be solved.
